model/channel_count.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2015 Spinpunch, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package model
+
+import (
+	"crypto/md5"
+	"encoding/json"
+	"fmt"
+	"io"
+	"sort"
+	"strconv"
+)
+
+type ChannelCounts struct {
+	Counts      map[string]int64 `json:"counts"`
+	UpdateTimes map[string]int64 `json:"update_times"`
+}
+
+func (o *ChannelCounts) Etag() string {
```

Comment:
The server code might not be faster but the the amount of bandwidth used to send the response will be lower and therefore be faster for clients to receive, which is particularly important for clients with slow connections. I'd definitely take a minor (and I believe this one is really minor) performance hit on the server to send back less data to the client.